Firestore/Source/API/FIRDocumentReference.m

```diff
@@ -236,14 +248,24 @@ - (void)getDocumentWithCompletion:(void (^)(FIRDocumentSnapshot *_Nullable docum
       //    offline.
       // 2) Actually call the completion handler with an error if the document doesn't exist when
       //    you are offline.
-      // TODO(dimond): Use proper error domain
       completion(nil,
                  [NSError errorWithDomain:FIRFirestoreErrorDomain
                                      code:FIRFirestoreErrorCodeUnavailable
                                  userInfo:@{
                                    NSLocalizedDescriptionKey :
                                        @"Failed to get document because the client is offline.",
                                  }]);
+    } else if (snapshot.exists && snapshot.metadata.fromCache &&
+               options.source == FIRSourceServer) {
+      completion(nil, [NSError errorWithDomain:FIRFirestoreErrorDomain
+                                          code:FIRFirestoreErrorCodeUnavailable
+                                      userInfo:@{
+                                        NSLocalizedDescriptionKey :
+                                            @"Failed to get document from server. (However, this "
+                                            @"document does exist in the local cache. Run again "
+                                            @"without setting FIRServer in the GetOptions to "
```

Comment:
This line mixes FIR-prefixed Server (which should be FIRSourceServer, incidentally) with non-prefixed GetOptions. These should be consistent (and probably prefixed).

Reply:
Done.